### PR TITLE
Go SDK updates

### PR DESCRIPTION
**Pull request by ameypv-rudder**
## What do these changes do?

> Updated the Go SDK with the latest changes.

## Type of documentation

- [ ] New documentation
- [x] Documentation update
- [ ] Hotfix

### CODE REVIEW

File: docs/sources/event-streams/sdks/rudderstack-go-sdk.mdx [hunk 1]
```diff
@@ -69,99 +76,118 @@ Alternatively, you can run the following snippet:
...
-            NoProxySupport:   true,
-        })
-```
+| **Field**      | **Type**      | **Description**        |
+| :---------------------- | :------------ |:---------------- |
+| `Interval`   | Time (in sec) | The SDK sends the messages when this flushing interval time has elapsed.       |
+| `BatchSize`       | Integer           | Total number of messages to be sent in a single batch.  Default value is 100.     |
+| `Verbose`  | Boolean  | When set to `true`, the client sends more frequent and detailed messages to the logger.  |
+| `RetryAfter`    | Time (in sec) |  Determines how long the SDK should wait to retry sending the failed messages. |
+| `maxConcurrentRequests` | Integer   | Maximum number of parallel goroutines to be spawned.          |
```
Comment (pallabmaiti) [inline docs/sources/event-streams/sdks/rudderstack-go-sdk.mdx:119]: Remove this.

File: docs/sources/event-streams/sdks/rudderstack-go-sdk.mdx [hunk 2]
```diff
@@ -69,99 +76,118 @@ Alternatively, you can run the following snippet:
...
-```go
-    // Instantiates a client to use send messages to the Rudder API.
-    client, _ := analytics.NewWithConfig(WRITE_KEY, DATA_PLANE_URL,
-        analytics.Config{
-            Interval:  30 * time.Second,
-            BatchSize: 100,
-            NoProxySupport:   true,
-        })
-```
+| **Field**      | **Type**      | **Description**        |
```
Comment (pallabmaiti) [inline docs/sources/event-streams/sdks/rudderstack-go-sdk.mdx:113]: Add Gzip in this table.

File: docs/sources/event-streams/sdks/rudderstack-go-sdk.mdx [hunk 3]
```diff
@@ -69,99 +76,118 @@ Alternatively, you can run the following snippet:
...
-| **Field**      | **Type** | **Presence**                              | **Description**                                                                                                            |
-| :------------- | :------- | :---------------------------------------- | :------------------------------------------------------------------------------------------------------------------------- |
-| `anonymousId`  | String   | Optional                                  | Sets the user ID for cases where there is no unique identifier for the user. Either `userId` or `anonymousId` is required. |
-| `userId`       | String   | Optional, if `anonymousId` is already set | Unique identifier for a particular user in your database.                                                                  |
-| `context`      | Object   | Optional                                  | Dictionary of information that provides context about a message. However, it is not directly related to the API call.      |
-| `integrations` | Object   | Optional                                  | A dictionary containing the destinations to be either enabled or disabled.                                                 |
```
Comment (pallabmaiti) [inline docs/sources/event-streams/sdks/rudderstack-go-sdk.mdx:152]: Don't remove `integrations` object.

File: docs/sources/event-streams/sdks/rudderstack-go-sdk.mdx [hunk 4]
```diff
@@ -170,19 +196,18 @@ client.Enqueue(analytics.Track{
...
-| `user_id`      | String | Required | The developer identification for your user                                                                                 |
-| `event`        | String | Required | Name of the event being performed by the user                                                                              |
-| `properties`   | Object | Optional | Dictionary of the properties associated with a particular event.                                                           |
-| `context`      | Object | Optional | Dictionary of information that provides context about a message. However, it is not directly related to the API call.      |
-| `timestamp`    | Date   | Optional | The timestamp of the message's arrival.                                                                                    |
-| `anonymous_id` | String | Optional | Sets the user ID for cases where there is no unique identifier for the user. Either `userId` or `anonymousId` is required. |
-| `integrations` | Object | Optional | A dictionary containing the destinations to be either enabled or disabled.                                                 |
```
Comment (pallabmaiti) [inline docs/sources/event-streams/sdks/rudderstack-go-sdk.mdx:181]: Don't remove `integrations` object.

File: docs/sources/event-streams/sdks/rudderstack-go-sdk.mdx [hunk 5]
```diff
@@ -197,25 +222,50 @@ client.Enqueue(analytics.Page{
...
-| **Field**      | **Type** | **Presence**                              | **Description**                                                                                                            |
-| :------------- | :------- | :---------------------------------------- | :------------------------------------------------------------------------------------------------------------------------- |
-| `anonymousId`  | String   | Optional                                  | Sets the user ID for cases where there is no unique identifier for the user. Either `userId` or `anonymousId` is required. |
-| `userId`       | String   | Optional, if `anonymousId` is already set | Unique identifier for a particular user in your database.                                                                  |
-| `context`      | Object   | Optional                                  | Dictionary of information that provides context about a message. However, it is not directly related to the API call.      |
-| `integrations` | Object   | Optional                                  | A dictionary containing the destinations to be either enabled or disabled.                                                 |
```
Comment (pallabmaiti) [inline docs/sources/event-streams/sdks/rudderstack-go-sdk.mdx:205]: Don't remove `integrations` object.

File: docs/sources/event-streams/sdks/rudderstack-go-sdk.mdx [hunk 6]
```diff
@@ -197,25 +222,50 @@ client.Enqueue(analytics.Page{
...
+  UserId: "1hKOmRA4GRlm",
+  Name:   "Pizza",
+  Properties: analytics.NewProperties().
+    SetURL("https://dominos.com"),
+})
+```
+
+The `screen` method parameters are as described below:
+
+| **Field**      | **Type** |  **Description**     |
```
Comment (pallabmaiti) [inline docs/sources/event-streams/sdks/rudderstack-go-sdk.mdx:251]: Add `integrations` object.

File: docs/sources/event-streams/sdks/rudderstack-go-sdk.mdx [hunk 7]
```diff
@@ -226,48 +276,41 @@ client.Enqueue(analytics.Group{
...
-| **Field**      | **Type** | **Presence**                              | **Description**                                                                                                            |
-| :------------- | :------- | :---------------------------------------- | :------------------------------------------------------------------------------------------------------------------------- |
-| `anonymousId`  | String   | Optional                                  | Sets the user ID for cases where there is no unique identifier for the user. Either `userId` or `anonymousId` is required. |
-| `userId`       | String   | Optional, if `anonymousId` is already set | Unique identifier for a particular user in your database.                                                                  |
-| `context`      | Object   | Optional                                  | Dictionary of information that provides context about a message. However, it is not directly related to the API call.      |
-| `integrations` | Object   | Optional                                  | A dictionary containing the destinations to be either enabled or disabled.                                                 |
```
Comment (pallabmaiti) [inline docs/sources/event-streams/sdks/rudderstack-go-sdk.mdx:234]: Don't remove `integrations` object.

File: docs/sources/event-streams/sdks/rudderstack-go-sdk.mdx [hunk 8]
```diff
@@ -226,48 +276,41 @@ client.Enqueue(analytics.Group{
...
 })
 ```
 
 The `alias` method parameters are as mentioned below:
 
-| **Field**      | **Type** | **Presence**                              | **Description**                                                                                                       |
-| :------------- | :------- | :---------------------------------------- | :-------------------------------------------------------------------------------------------------------------------- |
-| `userId`       | String   | Optional, if `anonymousId` is already set | Unique identifier for a particular user in your database.                                                             |
-| `context`      | Object   | Optional                                  | Dictionary of information that provides context about a message. However, it is not directly related to the API call. |
-| `integrations` | Object   | Optional                                  | A dictionary containing the destinations to be either enabled or disabled.                                            |
```
Comment (pallabmaiti) [inline docs/sources/event-streams/sdks/rudderstack-go-sdk.mdx:263]: Don't remove `integrations` object.

**Review by pallabmaiti — CHANGES_REQUESTED** [review]

File: docs/sources/event-streams/sdks/rudderstack-go-sdk.mdx (hunk 3 above)
Comment (ameypv-rudder) [reply docs/sources/event-streams/sdks/rudderstack-go-sdk.mdx:152]: Done, for all the instances.

File: docs/sources/event-streams/sdks/rudderstack-go-sdk.mdx [hunk 9]
```diff
@@ -14,48 +16,47 @@ Check out the [**GitHub codebase**](https://github.com/rudderlabs/analytics-go)
...
+2. <Link to="/dashboard-guides/sources/#adding-a-source">Set up a Go source</Link> in your dashboard. You should be able to see a <strong>write key</strong> for this source, as shown:
 
-<img src="../../../assets/event-stream-sources/go-write-key-new.png" alt="Go source write key" />
+<img src="../../../assets/event-stream-sources/go-sdk-write-key.png" alt="Go source write key" />
 
-- You will also need a data plane URL. Refer to the <Link to="/resources/glossary/#data-plane-url">Glossary</Link> for more information on the data plane URL and where to find it.
+You will also need a data plane URL. Refer to the <Link to="/dashboard-guides/overview/#data-plane-url">Dashboard Overview</Link> guide for more information on the data plane URL and where to find it.
 
 <div class="successBlock">
 The <strong>Setup</strong> tab in the RudderStack dashboard (seen above) has an SDK installation snippet containing both the write key and the data plane URL. You can use it to integrate the Go SDK into your application.
```
Comment (ishachopra0401) [inline docs/sources/event-streams/sdks/rudderstack-go-sdk.mdx:27]:
```suggestion
The <strong>Setup</strong> tab in the RudderStack dashboard (seen above) has the SDK installation snippet containing both the write key and the data plane URL. Copy it to integrate the Go SDK into your application.
```

File: docs/sources/event-streams/sdks/rudderstack-go-sdk.mdx [hunk 10]
```diff
@@ -69,99 +70,119 @@ Alternatively, you can run the following snippet:
...
-        })
-```
+| **Field**      | **Type**      | **Description**        |
+| :---------------------- | :------------ |:---------------- |
+| `Interval`   | Time (in sec) | The SDK sends the messages when this flushing interval time has elapsed.       |
+| `BatchSize`       | Integer           | Total number of messages to be sent in a single batch.  Default value is 100.     |
+| `Verbose`  | Boolean  | When set to `true`, the client sends more frequent and detailed messages to the logger.  |
+| `RetryAfter`    | Time (in sec) |  Determines how long the SDK should wait to retry sending the failed messages. |
+| `Gzip` | Integer   | Enables Gzip compression of the requests. By default, this is enabled (value set to `0`). To disable gzip, set this parameter to `1`.          |
+| `NoProxySupport`        | Boolean          | Setting this variable to `true` if you do not use a proxy to send the events.    |
```
Comment (ishachopra0401) [inline docs/sources/event-streams/sdks/rudderstack-go-sdk.mdx:114]:
```suggestion
| `NoProxySupport`        | Boolean          | Set this variable to `true` if you do not use a proxy to send the events.    |
```

File: docs/sources/event-streams/sdks/rudderstack-go-sdk.mdx [hunk 11]
```diff
@@ -69,99 +70,119 @@ Alternatively, you can run the following snippet:
...
-            NoProxySupport:   true,
-        })
-```
+| **Field**      | **Type**      | **Description**        |
+| :---------------------- | :------------ |:---------------- |
+| `Interval`   | Time (in sec) | The SDK sends the messages when this flushing interval time has elapsed.       |
+| `BatchSize`       | Integer           | Total number of messages to be sent in a single batch.  Default value is 100.     |
+| `Verbose`  | Boolean  | When set to `true`, the client sends more frequent and detailed messages to the logger.  |
+| `RetryAfter`    | Time (in sec) |  Determines how long the SDK should wait to retry sending the failed messages. |
+| `Gzip` | Integer   | Enables Gzip compression of the requests. By default, this is enabled (value set to `0`). To disable gzip, set this parameter to `1`.          |
```
Comment (ishachopra0401) [inline docs/sources/event-streams/sdks/rudderstack-go-sdk.mdx:113]:
```suggestion
| `Gzip` | Integer   | Enables gzip compression of the requests. By default, this is enabled (value set to `0`). To disable gzip, set this parameter to `1`.          |
```

File: docs/sources/event-streams/sdks/rudderstack-go-sdk.mdx [hunk 12]
```diff
@@ -69,99 +70,119 @@ Alternatively, you can run the following snippet:
...
 
 <div class="infoBlock">
+Setting <code class="inline-code">NoProxySupport</code> to <code class="inline-code">true</code> will avoid RudderStack making calls to the proxy for fetching the total number of nodes in case of a multi-node setup.
+</div>
 
-Setting <code class="inline-code">NoProxySupport</code> to <code class="inline-code">true</code> will avoid RudderStack making calls to the proxy for fetching the total number of nodes, in case of multi-node setup.
+## Gzipping requests
+
+<div class="successBlock">
+The Gzip feature is enabled by default in the Go SDK version 4.0.0.
```
Comment (ishachopra0401) [inline docs/sources/event-streams/sdks/rudderstack-go-sdk.mdx:123]:
```suggestion
The gzip feature is enabled by default in the Go SDK version 4.0.0.
```

**Review by ishachopra0401 — APPROVED** [review]

**Review by pallabmaiti — APPROVED** [review]

**Pull request merged**